comm/json_encoder.go

```diff
@@ -0,0 +1,47 @@
+package comm
+
+import (
+	"encoding/json"
+	"github.com/anycable/anycable-go/common"
+)
+
+type jsonEncoder struct {
+}
+
+func (jp jsonEncoder) Unmarshal(data []byte, message **common.Message) error {
+	return json.Unmarshal(data, message)
+}
+
+func (jp jsonEncoder) MarshalIsBinary() bool {
+	return false
+}
+
+func (jp jsonEncoder) MarshalReply(message *common.Reply) ([]byte, error) {
+	return json.Marshal(&message)
+}
+
+func (jp jsonEncoder) MarshalPing(message *common.PingMessage) ([]byte, error) {
+	return json.Marshal(&message)
+}
+
+func (jp jsonEncoder) MarshalDisconnect(message *common.DisconnectMessage) ([]byte, error) {
+	return json.Marshal(&message)
+}
+
+func (jp jsonEncoder) MarshalTransmissions(transmissions []string, message *common.Message) ([][]byte, error) {
+	var transmissionBytes [][]byte
+	for _, transmission := range transmissions {
+		transmissionBytes = append(transmissionBytes, []byte(transmission))
+	}
+
+	return transmissionBytes, nil
+}
+
+func (jp jsonEncoder) MarshalAuthenticateTransmissions(transmissions []string) ([][]byte, error) {
```

Comment:
Why do we need to treat these transmissions separately?

Reply:
Only to be able to provide the encoder with some context about the transmissions' origin. These are possible after authenticate reply (and there's no other context) or in a call reply (in which case there's a common.Message as context). That's why I split them into two separate API calls, one being called on authenticate reply, the other on call reply (with the message).
However, I just removed some duplicate code in the json encoder implementation.